### PR TITLE
Add support LabelME

### PR DESCRIPTION
**Pull request by kkdai**

### CODE REVIEW

**Review by narumiruna — APPROVED** [review]
LGTM

**Review by yoyo-pan — APPROVED** [review]

File: tools/labelme/labelme.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+type Shape struct {
+	Label     string  `bson:"label" json:"label"`
+	LineColor *[4]int `bson:"line_color" json:"line_color"`
+	FillColor *[4]int `bson:"fill_color" json:"fill_color"`
+	Points    []Point `bson:"points" json:"points"`
+}
+
+type Point [2]int
+
+func NewLabelME(imgPath string, label string, lineColor [4]int, fillColor [4]int, imgData []byte) *LabelME {
```
Comment (c9s) [inline tools/labelme/labelme.go]: when constructing object with more than 3 parameters, we should avoid using constructor function.

File: tools/labelme/labelme.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package labelme
+
+import (
+	"encoding/json"
+
+	"github.com/linkernetworks/annotation"
+)
+
+type LabelME struct {
```
Comment (c9s) [inline tools/labelme/labelme.go]: LabelmeJSON

File: tools/labelme/labelme.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+func NewLabelME(imgPath string, label string, lineColor [4]int, fillColor [4]int, imgData []byte) *LabelME {
+	return &LabelME{
+		ImagePath: imgPath,
+		LineColor: lineColor,
+		FillColor: fillColor,
+		ImageData: string(imgData),
+	}
+}
+
+func (l *LabelME) AddAnnotation(ann annotation.PolygonAnnotation, lineColor *[4]int, fillColor *[4]int) {
```
Comment (c9s) [inline tools/labelme/labelme.go]: we should avoid importing the "annotation" package. Please rename AddAnnotation to "AddShape" and only accept "Shape" type as the argument.

**Review by c9s — CHANGES_REQUESTED** [review]

File: tools/labelme/labelme.go (hunk 1 above)
Comment (zyfdegh) [reply tools/labelme/labelme.go]: any documentation to this rule?

File: tools/labelme/labelme.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+type Shape struct {
+	Label     string  `bson:"label" json:"label"`
+	LineColor *[4]int `bson:"line_color" json:"line_color"`
+	FillColor *[4]int `bson:"fill_color" json:"fill_color"`
+	Points    []Point `bson:"points" json:"points"`
+}
+
+type Point [2]int
+
+func NewLabelME(imgPath string, label string, lineColor [4]int, fillColor [4]int, imgData []byte) *LabelmeJSON {
```
Comment (c9s) [inline tools/labelme/labelme.go]: Use constructor literal instead of function

File: tools/labelme/labelme_test.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+		  255,
+		  0,
+		  0,
+		  128
+		],
+		"imagePath": "../Pictures/60168783_p0.png",
+		"imageData": "deadbeaf"
+	  }`
+
+	l := NewLabelME("../Pictures/60168783_p0.png", "aaa", [4]int{0, 255, 0, 128}, [4]int{255, 0, 0, 128}, []byte("deadbeaf"))
```
Comment (c9s) [inline tools/labelme/labelme_test.go]: avoid using upper case letter in the paths

File: tools/labelme/labelme_test.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+	ann := annotation.PolygonAnnotation{
+		Label:  "aaa",
+		Points: []annotation.Point{197, 210, 411, 187, 247, 323},
+	}
+	l.AddShape(ann, nil, nil)
```
Comment (c9s) [inline tools/labelme/labelme_test.go]: Pull out the annotation type -> Shape conversion to a function

**Review by c9s — CHANGES_REQUESTED** [review]

File: tools/labelme/labelme.go (hunk 1 above)
Comment (c9s) [reply tools/labelme/labelme.go]: the book "Clean Code" :-p

File: tools/labelme/labelme_test.go [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+		  255,
+		  0,
+		  0,
+		  128
+		],
+		"imagePath": "../pictures/60168783_p0.png",
+		"imageData": "deadbeaf"
+	  }`
+
+	l := NewLabelME("../pictures/60168783_p0.png", []byte("deadbeaf"))
```
Comment (c9s) [inline tools/labelme/labelme_test.go]:
```
l := LabelmeJSON{ 
   Field: "../pictures/60168783_p0.png",
   Field:   []byte("deadbeaf"),
 }
```

**Review by c9s — APPROVED** [review]

**Pull request merged**